### ISSUE TITLE
Python 3.7 regex _pattern_type

### ISSUE THREAD

**Issue by ajtucker** [sensiblecodeio/xypath#86]
I noticed an issue while running xypath under Python 3.7:

```
/usr/local/lib/python3.7/site-packages/xypath/xypath.py in filter(self, filter_by)
    404         elif have_ham and isinstance(filter_by, hamcrest.matcher.Matcher):
    405             return self._filter_internal(lambda cell: filter_by.matches(cell.value))
--> 406         elif isinstance(filter_by, re._pattern_type):
    407             return self._filter_internal(
    408                 lambda cell: re.match(filter_by, six.text_type(cell.value)))

AttributeError: module 're' has no attribute '_pattern_type'
```

It looks like re._pattern_type has been renamed to re.Pattern in 3.7, though there's also a newer way to check typing from Python 3.5 on, see https://docs.python.org/3.5/library/typing.html#typing.re.

**Comment by MikeData** [comment]
I'm getting the same issue

**Comment by StevenMaude** [comment]
@ajtucker Thanks for the report; sorry for this not being spotted so quickly.

Think this is fixed and will release to PyPI as 1.1.1, please let me know if it's not.
